File: IEEE2019/Xtreme_Rappers.cpp
#include <cstdlib>
#include <algorithm>
#include <iostream>
using namespace std;
typedef long long ll;

long long solve(ll K, ll J){
    ll alph = 2*J - K;
    if(alph%3LL != 0)
        return -1;
    ll bet = 2*K - J;
    if(bet%3LL != 0)
        return -1;
    return alph/3LL + bet/3LL;
}

ll srch(ll K, ll J){
    ll mx = 0;
    for(ll dj = 0; dj < 20; ++dj){
        for(ll dk = 0; dk < 20; ++dk){
            ll nk = K - dk;
            ll nj = J - dj;
            if(nk <= 0 || nj <= 0)
                continue;
            mx = max(mx, solve(nk,nj));
        }
    }
    return mx;
}

int main() {
    long long a, b;
    cin >> a >> b;
    long long cnt = 0;
    while ((a >= 2 && b >= 1) || (a >= 1 && b >=2)) {
        long long amount = 1;
        if(a == b){
            cnt += srch(a, b);
            a = 0;b = 0;
            break;
        }
        if (a > b) {
            long long diff = a - b;
            amount = diff / 2L > 1L ? diff / 2L : 1L;
            if (a - (2 * amount) < 0L) {
                amount = a / 2;
            }
            if ((b - amount) < 0L) {
                amount = b;
            }
            a -= (2L * amount);
            b -= (1L * amount);
        }
        else{
            long long diff = b - a;
            amount = diff / 2L > 1L ? diff / 2L : 1L;
            if (b - (2L * amount) < 0L) {
                amount = b / 2L;
            }
            if ((a - amount) < 0L) {
                amount = a;
            }
            b -= (2L * amount);
            a -= (1L * amount);
        }
        cnt += amount;
    }
    cout << cnt << endl;
    return 0;
}

File: IEEE2019/Xtreme_Rappers.py
def f(K, J, alpha, pr=False):
    A = K - 2*alpha
    B = J - alpha

    bet = min(A, B//2)

    if pr:
       return(alpha,bet)
    return alpha + bet

def srch(K, J):
    lo = 0
    hi = 10**15
    mx = 0
    while(hi - lo > 1):
        mid = (hi + lo)//2
        if (f(K, J, mid) > f(K, J, mid + 1)):
             hi = mid
        else:
             lo = mid

    a,b = f(K,J,lo+1,True)
    if a < 0 or b < 0:
        return 0
    return(max(0, f(K, J, lo+1)))

K, J = [int(x) for x in input().split()]
print(max(srch(K,J),srch(J,K)))
